codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -527,18 +538,26 @@ public ChannelFuture goAway(final ChannelHandlerContext ctx, final int lastStrea
 
             ChannelFuture future = frameWriter().writeGoAway(ctx, lastStreamId, errorCode, debugData, promise);
 
+            final String debugString = debugData.toString(UTF_8);
```

Comment:
@nmittler - Just do a retain and release then? Should be cheaper than a to string operation I think?